### ISSUE TITLE
Spurious "bounds [...] not enforced in type aliases" when using TAIT

### ISSUE THREAD

**Issue by Patryk27** [rust-lang/rust#108617]
### Code

```Rust
#![feature(type_alias_impl_trait)]

use std::fmt::Debug;

type Foo<T: Debug> = (impl Debug, usize);

fn foo<U: Debug>() -> Foo<U> {
    (Vec::<U>::new(), 1234)
}
```

### Current output

```
warning: bounds on generic parameters are not enforced in type aliases
 --> src/main.rs:5:13
  |
5 | type Foo<T: Debug> = (impl Debug, usize);
  |             ^^^^^
  |
  = note: `#[warn(type_alias_bounds)]` on by default
help: the bound will not be checked when the type alias is used, and should be removed
  |
5 - type Foo<T: Debug> = (impl Debug, usize);
5 + type Foo<T> = (impl Debug, usize);
  |
```

### Desired output

No warning at all, I think, since the bound _is_ actually checked; applying the suggestion yields code that does not compile.

**Comment by MU001999** [comment]
Yes, and the following code does not trigger the warning:
```rust
#![feature(type_alias_impl_trait)]

use std::fmt::Debug;

type Foo<T: Debug> = impl Debug;

fn foo<U: Debug>() -> Foo<U> {
    Vec::<U>::new()
}
```

**Comment by MU001999** [comment]
Currently, this lint passes `type X = impl Trait`. It seems like there shouldn't be a warning if the type has any opaque types.

**Comment by oli-obk** [comment]
In https://github.com/rust-lang/rust/pull/108663 we figured out that we could add `ty::AliasKind::Alias` which could then represent type aliases properly. But instead of converting all type aliases, which has its own problems, we just do it for type aliases with opaque types, which already have all the restrictions needed to just solve this issue.
